plugins/core/src/terminal-service.ts

```diff
@@ -145,6 +145,17 @@ export class TerminalService extends ScryptedDeviceBase implements StreamService
      *   EOF: { "eof": true }
      */
     async connectStream(input: AsyncGenerator<Buffer | string, void>, options?: any): Promise<AsyncGenerator<Buffer, void>> {
+        if (options?.clusterWorkerId) {
+            const clusterWorkerId = options.clusterWorkerId;
+            delete options.clusterWorkerId;
+
+            const fork = sdk.fork<{
+                connectTTYStream: typeof connectTTYStream,
+            }>({ clusterWorkerId });
+            const result = await fork.result;
+            return result.connectTTYStream(input, options);
```

Comment:
if fork.result throws, or connectTTYstream throws, I would also call fork.terminate to force immediate cleanup.